### PR TITLE
add job to sync app log metadata from garden to silk datastore.

### PR DESCRIPTION
**Pull request by Benjamintf1**
Updates silk datastore so that silk will refelect app renames correctly.

### CODE REVIEW

**Comment by Benjamintf1** [comment]
@ctlong mentioned to look into permissions on the silk-datastore-syncer and packaging spec. I'll try and make sure to look into that.

File: jobs/silk-datastore-syncer/templates/bpm.yml.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+processes:
+  - name: silk-datastore-syncer
+    unsafe:
+      privileged: true
```
Comment (ctlong) [inline jobs/silk-datastore-syncer/templates/bpm.yml.erb:5]: Do we need this privileged setting? Or the volumes and capabilities outlined below?

File: packages/silk-datastore-syncer/spec [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+name: silk-datastore-syncer
+
+dependencies:
+  - golang-1.19-linux
+
+files:
+  - code.cloudfoundry.org/**/*
```
Comment (ctlong) [inline packages/silk-datastore-syncer/spec]: The rest of the silk packages are more specific in identifying which files are necessary, maybe we should be too?

**Comment by ctlong** [comment]
> @ctlong mentioned to look into permissions on the silk-datastore-syncer and packaging spec. I'll try and make sure to look into that.

Sorry, I didn't see your msg before I submitted the feedback. Was just trying to do the same thing as you and note the stuff to look into on the PR 😁

File: packages/silk-datastore-syncer/spec (hunk 2 above)
Comment (geofffranks) [reply packages/silk-datastore-syncer/spec]: Yeah we should stay consistent with the rest of the release. We'll add it to the sync-package specs script + get this fixed after merging. thanks for calling this out!

File: jobs/silk-datastore-syncer/templates/bpm.yml.erb (hunk 1 above)
Comment (geofffranks) [reply jobs/silk-datastore-syncer/templates/bpm.yml.erb:5]: You'll need the volumes in order to modify the metadata file + access the garden socket when it's in unix mode. I'm not sure about network capabilities or privileged mode though.

If these were cargo culted from another job, can you try removing the capabilities + privileged mode, and run this against a garden with `garden.listen_network` set to `tcp` and also with it set to `unix` to make sure it works in both cases?

Comment (Benjamintf1) [reply jobs/silk-datastore-syncer/templates/bpm.yml.erb:5]: do we even still support tcp listen network anymore?

**Comment by geofffranks** [comment]
It’s still supported in the boshrelease for Linux, and also used on Windows. We also should probably test it when garden runs in bpm-enabled mode as well.

From: Benjamin Fuller ***@***.***>
Date: Friday, March 31, 2023 at 12:25 PM
To: cloudfoundry/silk-release ***@***.***>
Cc: Geoff Franks ***@***.***>, Comment ***@***.***>
Subject: Re: [cloudfoundry/silk-release] add job to sync app log metadata from garden to silk datastore. (PR #75)
!! External Email

@Benjamintf1 commented on this pull request.

________________________________

In jobs/silk-datastore-syncer/templates/bpm.yml.erb<https://github.com/cloudfoundry/silk-release/pull/75#discussion_r1154675469>:

> @@ -0,0 +1,14 @@

+---

+processes:

+  - name: silk-datastore-syncer

+    unsafe:

+      privileged: true

do we even still support tcp listen network anymore?

—
Reply to this email directly, view it on GitHub<https://github.com/cloudfoundry/silk-release/pull/75#discussion_r1154675469>, or unsubscribe<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FABWZEUX2ID2LY7YK4P4Z2PTW64APNANCNFSM6AAAAAAWMM4RSI&data=05%7C01%7Cgfranks%40vmware.com%7C31571b57c2dc4a47c4e308db32048996%7Cb39138ca3cee4b4aa4d6cd83d9dd62f0%7C0%7C0%7C638158767326804531%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=H9w3%2By6eaXqyDVix1Y9Kq5M6nTypRQlY65MupiHoLc0%3D&reserved=0>.
You are receiving this because you commented.Message ID: ***@***.***>

!! External Email: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender.

File: jobs/silk-datastore-syncer/templates/bpm.yml.erb (hunk 1 above)
Comment (Benjamintf1) [reply jobs/silk-datastore-syncer/templates/bpm.yml.erb:5]: image

I managed to remove this, but couldn't get tcp mode to work.

**Comment by Benjamintf1** [comment]
I updated the package specs.

**Comment by Benjamintf1** [comment]
There's something else I want to check (race conditions things), but if the other jobs have these permissions and if they're too much, is that something we should worry about for this pr or possibly take on in some other more cross cutting effort?

File: src/code.cloudfoundry.org/silk-datastore-syncer/main.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+			}
+
+			logger.Debug("Datastore container reconciling with Garden container", lager.Data{"handle": sc.Handle})
+			b, err := json.Marshal(desiredLogConfig)
+			if err != nil {
+				logger.Error("Garden container error marshalling container log config", err, lager.Data{"handle": sc.Handle})
+				continue
+			}
+			sc.Metadata["log_config"] = string(b)
+			err = store.Add(sc.Handle, sc.IP, sc.Metadata)
```
Comment (Benjamintf1) [inline src/code.cloudfoundry.org/silk-datastore-syncer/main.go]: This should maybe be a update. Could cause re-adding things on accident and leave limbo entries.

Comment (Benjamintf1) [reply src/code.cloudfoundry.org/silk-datastore-syncer/main.go]: from my perspective these tests are completely unrunnable tho as is. They just fail.

**Comment by Benjamintf1** [comment]
race conditions should be managed.

**Review by geofffranks — APPROVED** [review]

**Pull request merged**